compiler/GHC/Hs/Expr.hs

```diff
@@ -2332,7 +2333,7 @@ pprArg (ApplicativeArgOne _ pat expr isBody)
 pprArg (ApplicativeArgMany _ stmts return pat) =
      ppr pat <+>
      text "<-" <+>
-     ppr (HsDo (panic "pprStmt") DoExpr (noLoc
+     ppr (HsDo (panic "pprStmt") (DoExpr Nothing) (noLoc
```

Comment:
It is likely that we should get the context from elsewhere. But I think this only affects error messages, and this doesn't block the other tasks we have depending on this PR.

Unless you have any hints on how to get GHC to use this code, I will investigate it and fix it in another PR.

Reply:
Yes, I believe `ppr` is only used for internal messages and errors.
However, I find it a bit suspcious that we do not touch `ApplicativeArgMany`, I suspect we'll need to do so later to make sure the interaction with `ApplicativeDo` is smooth..

Reply:
`ApplicativeArgMany` is not constructed by the parser but by `stmtTreeToStmts` in the renamer. I'm creating a new issue to implement this once we can test errors with QualifiedDo and ApplicativeDo.